File: design-hit-counter/design-hit-counter.js
/**
 * Initialize your data structure here.
 */
var HitCounter = function() {
    this.q = [];
    this.time = 300;
};

/**
 * Record a hit.
        @param timestamp - The current timestamp (in seconds granularity). 
 * @param {number} timestamp
 * @return {void}
 */
HitCounter.prototype.hit = function(t) {
    this.q.push(t)
};

/**
 * Return the number of hits in the past 5 minutes.
        @param timestamp - The current timestamp (in seconds granularity). 
 * @param {number} timestamp
 * @return {number}
 */
HitCounter.prototype.getHits = function(t) {
    let c = 0;
    
    let start= Math.max(0, t-300)
    for(let i = 0; i < this.q.length; i++){
        if(this.q[i] > start && this.q[i] <= t){
            console.log("Asjhish")
            c++
        }
    }
    
    return c
};

/** 
 * Your HitCounter object will be instantiated and called as such:
 * var obj = new HitCounter()
 * obj.hit(timestamp)
 * var param_2 = obj.getHits(timestamp)
 */

File: design-hit-counter/design-hit-counter.py
class HitCounter:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.count = [0] * 300
        self.hits = [0]*300
        

    def hit(self, timestamp: int) -> None:
        """
        Record a hit.
        @param timestamp - The current timestamp (in seconds granularity).
        """
        key = timestamp % 300
        
        if self.hits[key] != timestamp:
            self.hits[key] = timestamp
            self.count[key] = 1
        else:
            self.count[key] += 1
        

    def getHits(self, timestamp: int) -> int:
        """
        Return the number of hits in the past 5 minutes.
        @param timestamp - The current timestamp (in seconds granularity).
        """
        
        ans = 0
        
        for i in range(300):
            if self.hits[i] > timestamp - 300:
                ans += self.count[i]
                
        return ans
        


# Your HitCounter object will be instantiated and called as such:
# obj = HitCounter()
# obj.hit(timestamp)
# param_2 = obj.getHits(timestamp)